### PR TITLE
Represent `protocol` within `wanlib` and update go-webauthn

### PR DESCRIPTION
**Pull request by codingllama**
Represent the entirety of go-webauthn's protocol within `lib/auth/webauthn`, instead of only "leaf" types, so we are isolated from library JSON marshal changes (which there are plenty).

This allows us to update go-webauthn without potential compatibility breakages between Teleport versions, pushing us all the way the latest dependency. (We don't care whether bytes are Standard- or RawURL-encoded, transport is settled for us.)

Tested using pre- and post-update `tsh` binaries.

### CODE REVIEW

**Review by tobiaszheller — APPROVED** [review]
lgtm, can you also check it with new and old version of server?

**Review by jentfoo — APPROVED** [review]

**Review by rosstimothy — COMMENTED** [review]
> Tested using pre- and post-update tsh binaries.

What was the version matrix that was tested? Did we test new/old tsh against new/old Auth?

File: lib/auth/webauthn/messages.go [hunk 1]
```diff
@@ -15,12 +15,35 @@
...
 	"github.com/gravitational/trace"
 )
 
 // CredentialAssertion is the payload sent to authenticators to initiate login.
-type CredentialAssertion protocol.CredentialAssertion
+type CredentialAssertion struct {
+	Response PublicKeyCredentialRequestOptions `json:"publicKey"`
+}
+
+type PublicKeyCredentialRequestOptions struct {
```
[the remark this answers is not in the document]
Comment (codingllama) [reply lib/auth/webauthn/messages.go:31]: I wasn't planning on documenting non-top-level messages - copying docs from go-webauthn seems unappealing and docs like "Foo is a clone of [protocol.Foo]", which I had at one point, don't really add anything.

File: lib/auth/webauthn/messages.go [hunk 2]
```diff
@@ -52,9 +111,59 @@ type CredentialAssertionResponse struct {
 	AssertionResponse AuthenticatorAssertionResponse `json:"response"`
 }
 
+type AuthenticatorAssertionResponse struct {
```
[the remark this answers is not in the document]
Comment (codingllama) [reply lib/auth/webauthn/messages.go:114]: Ack.

**Comment by codingllama** [comment]
> lgtm, can you also check it with new and old version of server?

> What was the version matrix that was tested? Did we test new/old tsh against new/old Auth?

I tested old and new tsh vs new server. The matrix goes somewhat like this:

||server old|server new|
|-|-|-|
|tsh old|no need|✅
|tsh new|not tested|✅

I can cover "tsh new" + "server old", just in case. I'll update the PR with the results, although I don't expect any surprises.

**Comment by codingllama** [comment]
Oh, just to add a bit more detail, I tested both with carried over credentials (MFA, Passwordless and Touch ID) and then tested deleting/adding new credentials with the "new"+"new" combination, just to be sure.

**Comment by codingllama** [comment]
PTAL, test matrix complete.

**Comment by codingllama** [comment]
Friendly ping @rosstimothy ?

File: lib/auth/webauthn/messages.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (codingllama) [reply lib/auth/webauthn/messages.go:31]: There's little chance you'd navigate WebAuthn without doing some background reading first. I can add the boilerplate comments back, but I won't copy docs from `protocol` or thoroughly explain all of them, I don't think it's worth it.

**Review by rosstimothy — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

**Pull request merged**